homework/tai_nguyen/239_sliding_window_maximum.ts

```diff
@@ -0,0 +1,37 @@
+function front(dequeue: number[]) {
+  if (dequeue.length) {
+    return dequeue[0];
+  }
+
+  return undefined;
+}
+
+function back(dequeue: number[]) {
+  if (dequeue.length) {
+    return dequeue[dequeue.length - 1];
+  }
+
+  return undefined;
+}
+
+function maxSlidingWindow(nums: number[], k: number): number[] {
+  const res: number[] = [];
+  const dequeue: number[] = [];
+
+  for (let i = 0; i < nums.length; i++) {
+    while (dequeue.length && nums[back(dequeue)!] < nums[i]) {
```

Comment:
I've never seen the syntax `nums[back(dequeue)!]`, I mean the (!) notation
Can you explain about it for me ? any document about it would be really helpful :🙇‍♂️

Reply:
Hi @tadashii1417, `!` is non-null assertion operator supported by Typescript >= v2.0 and it is not available in Javascript.
Futher info: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator